src/stores/accountDetails.js

```diff
@@ -86,6 +102,43 @@ export const useAccountStore = defineStore('account', () => {
     rawAccountNames.value = data
   }
 
+  async function fetchAccountTokens({ accountId, queryParameters, limit } = {}) {
+    rawAccountTokens.value = null
+    const defaultParameters = `/v2/aex9/account-balances/${accountId}?limit=${limit ?? 10}`
+    const { data } = await axios.get(`${MIDDLEWARE_URL}${queryParameters || defaultParameters}`)
+    rawAccountTokens.value = data
+
+    if (rawAccountTokens.value?.data.length) {
+      await fetchAccountTokensPrices()
+    }
+  }
+
+  async function fetchAccountTokensPrices() {
```

Comment:
I think this part can be improved. It's not needed to fetch the price for every single token. Right now there are 3 (+ Lexon) listed tokens which can be retrieved from here:
https://dex-backend-mainnet.prd.aepps.com/tokens/listed

It might make sense to cache this endpoint and retrieve the price only if a token is part of this list, this will save a lot of requests for every row.

Reply:
@michele-franchi can I split this optimization into a separate task? When implementing your recommendation it would be best to refactor also the tokens and token details so it's cached all across the application, not this specific place. But this is out of the initial scope making the task bigger than the 1 point we estimated.

Reply:
Ok, let's do it.